helm/metac/crds/metac.openebs.io_genericcontrollers.yaml

```diff
@@ -348,21 +347,8 @@ spec:
                                 - values
                                 type: object
                               type: array
-                          required:
-                          - matchAnnotationExpressions
-                          - matchAnnotations
-                          - matchFieldExpressions
-                          - matchFields
-                          - matchLabelExpressions
-                          - matchLabels
-                          - matchReference
-                          - matchReferenceExpressions
-                          - matchSlice
-                          - matchSliceExpressions
                           type: object
                         type: array
-                    required:
```

Comment:
If I understand correctly, we change CRD definition here. It should probably cause apiVersion to change to have backward compatibility for API

Reply:
The schema fields has been changed from being mandatory to optional. It should not break ideally since earlier they were all mandatory. In addition, the logic/code behind treated these fields to be optional from the start.

In an ideal world, it makes sense to change the apiVersion. However, doing so makes things much more unstable & requires transformation & other tooling to upgrade the CRDs. In short, things have not matured in k8s tooling to rollout apiVersion related changes to k8s environments.

Reply:
I agree that in this case this might be not required. 

Not sure I get arguments about rolling out api versions : As far I understand, this is common practice in kubernetes (the API is evolving) and k8s is ready to handle more then one API version.

Resolving thread, but happy to discuss on slack your remarks about API versioning :)

Reply:
I don't think this needs an api version bump it's not a braking change, if we were making things mandatory that would be a different case.